geodiff/src/geodiffrebase.cpp

```diff
@@ -321,326 +284,230 @@ int _find_mapping_for_new_changeset( const Buffer &buf,
 }
 
 
-void _handle_insert(
-  Sqlite3ChangesetIter &pp,
-  const std::string tableName,
-  int pnCol,
-  const RebaseMapping &mapping,
-  unsigned char *aiFlg,
-  std::shared_ptr<BinaryStream> out
-)
+bool _handle_insert( const ChangesetEntry &entry, const RebaseMapping &mapping, ChangesetEntry &outEntry )
 {
-  // first write operation type (iType)
-  out->put( SQLITE_INSERT );
-  out->put( 0 );
+  size_t numColumns = entry.table->primaryKeys.size();
```

Comment:
number of columms is number of primary keys?

Reply:
primaryKeys array is an array of bools - each column is true/false depending on whether it is part of pkey or not. So the size of the array == number of columns.

Reply:
would it be possible to create new function in entry.table to return number of columns. would be much more obvious